ultra/ultra/scenarios/common/begin_time_init_funcs.py

```diff
@@ -71,16 +76,37 @@ def burst_begin_time_init_func(
             vehicle_spawn_time += time_so_far
             _begin_times.extend(vehicle_spawn_time)
 
-            time_to_next_cluster = np.random.uniform(*time_between_cluster)
-            time_so_far += time_to_next_cluster
+            speed = "".join(filter(lambda i: i.isdigit(), speed))
+            if density == "high-density":
+                if int(speed) < 70:
+                    variable_time_between_cluster = (
+                        time_between_cluster[0] + speed_offset + (num_lanes / 2),
+                        time_between_cluster[1] + speed_offset + (num_lanes / 2),
+                    )
+                elif int(speed) > 70:
+                    variable_time_between_cluster = (
+                        time_between_cluster[0] - speed_offset - (num_lanes / 2),
+                        time_between_cluster[1] - speed_offset - (num_lanes / 2),
+                    )
+                else:
+                    variable_time_between_cluster = (
+                        time_between_cluster[0],
+                        time_between_cluster[1],
+                    )
```

Comment:
Actually, I was thinking about this again, and just so that we don't run into any problems when/if we ever add any other speeds to ULTRA, perhaps we can raise an exception if we encounter a speed we don't currently have. Maybe something like:
```suggestion
                if int(speed) == 50:
                    variable_time_between_cluster = (
                        time_between_cluster[0] + speed_offset + (num_lanes / 2),
                        time_between_cluster[1] + speed_offset + (num_lanes / 2),
                    )
                elif int(speed) == 70:
                    variable_time_between_cluster = (
                        time_between_cluster[0],
                        time_between_cluster[1],
                    )
                elif int(speed) == 100:
                    variable_time_between_cluster = (
                        time_between_cluster[0] - speed_offset - (num_lanes / 2),
                        time_between_cluster[1] - speed_offset - (num_lanes / 2),
                    )
                else:
                    raise ValueError(f"Unsupported speed: {speed}.")
```
Let me know your thoughts.

Reply:
Yes, I agree. At least from what I can see that a new speed can only be inducted if and only if a new map is constructed